Readability.js

```diff
@@ -245,10 +260,24 @@ Readability.prototype = {
       if (curTitle.split(' ').length < 3)
         curTitle = origTitle.replace(/[^\|\-]*[\|\-](.*)/gi,'$1');
     } else if (curTitle.indexOf(': ') !== -1) {
-      curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+      // XXX check if we have an heading containing this exact string, so we
+      // could assume it's the full title
+      // var headings = [].concat.call([].slice.call(doc.getElementsByTagName('h1')),
+      //                               [].slice.call(doc.getElementsByTagName('h2')));
+      var headings = this._concatNodeLists(
+        doc.getElementsByTagName('h1'),
+        doc.getElementsByTagName('h2')
+      );
+      var match = this._someNode(headings, function(heading) {
+        return heading.textContent === curTitle;
+      });
 
-      if (curTitle.split(' ').length < 3)
-        curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
+      if (!match) {
+        curTitle = origTitle.replace(/.*:(.*)/gi, '$1');
+
+        if (curTitle.split(' ').length < 3)
+          curTitle = origTitle.replace(/[^:]*[:](.*)/gi,'$1');
```

Comment:
or, in fact, using indexOf/lastIndexOf, as we're already guaranteed there's at least one colon if we're even in this block.